src/components/SessionIncrementor.tsx

```diff
@@ -0,0 +1,15 @@
+import { useEffect, memo } from 'react';
+
+import { useDispatch } from 'react-redux';
+
+import { incrementSessionCount } from 'src/redux/slices/session';
+
+const SessionIncrementor = () => {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(incrementSessionCount());
+  }, [dispatch]);
```

Comment:
1. This part unfortunately, depends heavily on hooks `useDispatch` and `useEffect`. That's basically all it does.

So there's almost 0 logic we can extract into a util function

2. We cannot put this in _app.tsx. Since this hook need to be under `ReduxProvider`. (Cannot be called outside `ReduxProvider`). So it need to be inside another component.

- Similar pattern is also implemented in `GlobalScrollListener` (which actually can be just a hook inside `GlobalListener` ), and `AudioRepeatManager`. Which literally renders nothing. 
- I am not a big fan of this pattern, but I am following this for consistency